### PR TITLE
Implementing torch.isfinite

### LINKED ISSUE #9132: [feature request] torch.isinf, torch.isfinite
Exists in numpy: https://docs.scipy.org/doc/numpy/reference/generated/numpy.isinf.html (https://docs.scipy.org/doc/numpy/reference/generated/numpy.isfinite.html) and tensorflow https://www.tensorflow.org/api_docs/python/tf/is_inf (https://www.tensorflow.org/api_docs/python/tf/is_finite)

Useful for debugging checks like `assert not (torch.isnan(x).any() or torch.isinf(x).any())`. Much more concise than `assert not (torch.isnan(x).any() or x.eq(float('inf')).any() or x.eq(float('-inf')).any())`

### PR DESCRIPTION
**Pull request by bhushan23**
fixes #9132

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -4707,14 +4707,18 @@ def test_logical(self):
         self.assertEqual(neqs.long().sum(), xne.long().sum(), 0)
         self.assertEqual(x.nelement(), all.long().sum())
 
-    def test_isnan(self):
-        x = torch.Tensor([1, float('nan'), 2])
-        self.assertEqual(torch.isnan(x), torch.ByteTensor([0, 1, 0]))
+    def test_isfinite(self):
+        x = torch.Tensor([1, float('inf'), 2, float('-inf'), float('nan')])
+        self.assertEqual(torch.isinf(x), torch.ByteTensor([1, 0, 1, 0, 0]))
```
Comment (vishwakftw) [inline test/test_torch.py]: This test is wrong.

**Review by zou3519 — APPROVED** [review]

File: torch/functional.py [hunk 2]
```diff
@@ -136,6 +137,25 @@ def btriunpack(LU_data, LU_pivots, unpack_data=True, unpack_pivots=True):
...
+        Tensor: A ``torch.ByteTensor`` containing a 1 at each location of finite elements and 0 otherwise
+
+    Example::
+
+        >>> torch.isfinite(torch.Tensor([1, float('inf'), 2, float('-inf'), float('nan')]))
+        tensor([ 1,  0,  1,  0,  0], dtype=torch.uint8)
+    """
+    if not isinstance(tensor, torch.Tensor):
+        raise ValueError("The argument is not a tensor", str(tensor))
+    return (isinf(tensor) | isnan(tensor)) == 0
```
Comment (SsnL) [inline torch/functional.py]: This should be more efficient as `(tensor == tensor) & (tensor.abs() != math.inf)`.

**Review by ssnl — CHANGES_REQUESTED** [review]

**Comment by bhushan23** [comment]
Thanks @zou3519 @vishwakftw for the review
@SsnL your suggested is indeed efficient 4 ops vs earlier 5 ops - updated.

**Review by ssnl — APPROVED** [review]

**Comment by bhushan23** [comment]
@ezyang approval required!!

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**